test/config-luatest/rpc_test.lua

```diff
@@ -419,6 +438,114 @@ g.test_filter_mode = function(g)
+    local config = [[
+    credentials:
+      users:
+        guest:
+          roles: [super]
+        myuser:
+          password: "secret"
+          roles: [replication]
+          privileges:
+          - permissions: [execute]
+            universe: true
+
+    roles: [one]
+
+    iproto:
+      listen:
+        - uri: 'unix/:./{{ instance_name }}.iproto'
+      advertise:
+        peer:
+          login: 'myuser'
+
+    groups:
+      group-001:
+        replicasets:
+          replicaset-001:
+            instances:
+              instance-001:
+                database:
+                  mode: rw
+          replicaset-002:
+            instances:
+              instance-002: {}
+              instance-003: {}
+              instance-004: {}
+              instance-005: {}
+    ]]
+    treegen.write_file(dir, 'config.yaml', config)
+
+    local role = string.dump(function()
+        local function f1()
+            return box.info.name
+        end
+
+        rawset(_G, 'f1', f1)
+
+        return {
+            stop = function() end,
+            apply = function() end,
+            validate = function() end,
+        }
+    end)
+    treegen.write_file(dir, 'one.lua', role)
+
+    local opts = {
+        env = {LUA_PATH = os.environ()['LUA_PATH']},
+        config_file = 'config.yaml',
+        chdir = dir,
+    }
+    g.server_1 = server:new(fun.chain(opts, {alias = 'instance-001'}):tomap())
+
+    g.server_1:start({wait_until_ready = false})
+    start_stub_servers(g, dir, {
+        'instance-002',
+        'instance-003',
+        'instance-004',
+        'instance-005'
+    })
+
+    g.server_1:wait_until_ready()
+
+    local function check_filter()
+        local connpool = require('experimental.connpool')
+        local clock = require('clock')
+
+        -- The connection timeout for filter() is hardcoded in
+        -- connpool and equals to 10 seconds.
+        local CONNECT_TIMEOUT = 10
+
+        -- If the connection pool tries to connect in parallel, the
+        -- execution time is bounded with CONNECT_TIMEOUT plus some
+        -- small overhead.
+        --
+        -- Make sure we're trying to access the instances in parallel.
+        local opts = { mode = 'ro' }
+        local timestamp_before_filter = clock.monotonic()
+        t.assert_equals(connpool.filter(opts), {})
+        local elapsed_time = clock.monotonic() - timestamp_before_filter
+        t.assert_gt(elapsed_time, CONNECT_TIMEOUT / 2)
+        t.assert_lt(elapsed_time, CONNECT_TIMEOUT * 3 / 2)
+
+        -- call() also internally uses filter(). Ensure it tries
+        -- to access instances in parallel.
+        opts = { mode = 'prefer_ro' }
+        timestamp_before_filter = clock.monotonic()
+        t.assert_equals(connpool.call('f1', nil, opts), 'instance-001')
+        elapsed_time = clock.monotonic() - timestamp_before_filter
+        t.assert_gt(elapsed_time, CONNECT_TIMEOUT / 2)
+        t.assert_lt(elapsed_time, CONNECT_TIMEOUT * 3 / 2)
```

Comment:
Should we separate the filter/call test cases (see the idea [here](https://github.com/tarantool/tarantool/pull/10438#discussion_r1774266075))?

Reply:
Done. Thank you.